### PR TITLE
fix(components): [tree] setCheckedNodes incorrectly checked node

### LINKED ISSUE #15375: [Component] [tree] el-tree 有禁用节点，调用setCheckedNodes无法选中未禁用节点
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.4`
- Element Plus Version: `2.4.4`
- Browser / OS: `Chrome 120.0.6099.200`
- Build Tool: `Vite`

### Related Component

- `All`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiA8ZWwtY2hlY2tib3hcbiAgICAgICAgICB2LW1vZGVsPVwiY2hlY2tlZFwiXG4gICAgICAgICAgQGNoYW5nZT1cImNoZWNrZWRBbGxcIlxuICAgICAgICAgID7lhajpgIlcbiAgICAgICAgPC9lbC1jaGVja2JveD5cbiAgPGVsLXRyZWUgICAgICBcbiAgICAgIG5vZGUta2V5PVwiaWRcIlxuICAgICAgIHJlZj1cInRyZWVSZWZcIiBcbiAgICAgICA6ZGF0YT1cImRhdGFcIiA6cHJvcHM9XCJkZWZhdWx0UHJvcHNcIiBzaG93LWNoZWNrYm94IC8+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBkZWZhdWx0UHJvcHMgPSB7XG4gIGNoaWxkcmVuOiAnY2hpbGRyZW4nLFxuICBsYWJlbDogJ2xhYmVsJyxcbiAgZGlzYWJsZWQ6ICdkaXNhYmxlZCcsXG59XG5jb25zdCBjaGVja2VkID0gcmVmKGZhbHNlKVxuY29uc3QgaXNJbmRldGVybWluYXRlID0gcmVmKGZhbHNlKVxuY29uc3QgdHJlZVJlZiA9IHJlZihudWxsKVxuXG5jb25zdCBkYXRhID0gW1xuICB7XG4gICAgaWQ6IDEsXG4gICAgbGFiZWw6ICdMZXZlbCBvbmUgMScsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgaWQ6IDMsXG4gICAgICAgIGxhYmVsOiAnTGV2ZWwgdHdvIDItMScsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAge1xuICAgICAgICAgICAgaWQ6IDQsXG4gICAgICAgICAgICBsYWJlbDogJ0xldmVsIHRocmVlIDMtMS0xJyxcbiAgICAgICAgICB9LFxuICAgICAgICAgIHtcbiAgICAgICAgICAgIGlkOiA1LFxuICAgICAgICAgICAgbGFiZWw6ICdMZXZlbCB0aHJlZSAzLTEtMicsXG4gICAgICAgICAgICBkaXNhYmxlZDogdHJ1ZSxcbiAgICAgICAgICB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgaWQ6IDIsXG4gICAgICAgIGxhYmVsOiAnTGV2ZWwgdHdvIDItMicsXG4gICAgICAgIGRpc2FibGVkOiB0cnVlLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGlkOiA2LFxuICAgICAgICAgICAgbGFiZWw6ICdMZXZlbCB0aHJlZSAzLTItMScsXG4gICAgICAgICAgfSxcbiAgICAgICAgICB7XG4gICAgICAgICAgICBpZDogNyxcbiAgICAgICAgICAgIGxhYmVsOiAnTGV2ZWwgdGhyZWUgMy0yLTInLFxuICAgICAgICAgICAgZGlzYWJsZWQ6IHRydWUsXG4gICAgICAgICAgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbl1cblxuZnVuY3Rpb24gZmxhdHRlblRyZWUoZGF0YSkge1xuICBjb25zdCByZXN1bHQgPSBbXVxuXG4gIGZ1bmN0aW9uIGZsYXR0ZW4obm9kZSkge1xuICAgIHJlc3VsdC5wdXNoKG5vZGUpXG4gICAgaWYgKG5vZGU/LmNoaWxkcmVuPy5sZW5ndGgpIHtcbiAgICAgIG5vZGUuY2hpbGRyZW4uZm9yRWFjaChmbGF0dGVuKVxuICAgIH1cbiAgfVxuXG4gIGRhdGEuZm9yRWFjaChmbGF0dGVuKVxuICByZXR1cm4gcmVzdWx0XG59XG5cbmNvbnN0IGNoZWNrZWRBbGwgPSAoYm9vbCkgPT4ge1xuICBjb25zb2xlLmxvZyhib29sLCAnY2hlY2tlZEFsbCcpXG4gIC8vIOiuvue9ruWLvumAieiKgueCuSAg5bmz6ZO655qE5pWw5o2uXG4gIHRyZWVSZWYudmFsdWUuc2V0Q2hlY2tlZE5vZGVzKGJvb2wgPyBmbGF0dGVuVHJlZShkYXRhKS5maWx0ZXIoaXRlbSA9PiAgIWl0ZW0uZGlzYWJsZWQgKTogW10pXG59XG48L3NjcmlwdD5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

<template>
 <el-checkbox
          v-model="checked"
          @change="checkedAll"
          >全选
        </el-checkbox>
  <el-tree      
      node-key="id"
       ref="treeRef" 
       :data="data" :props="defaultProps" show-checkbox />
</template>

<script setup>
import { ref } from 'vue'

const defaultProps = {
  children: 'children',
  label: 'label',
  disabled: 'disabled',
}
const checked = ref(false)
const isIndeterminate = ref(false)
const treeRef = ref(null)

const data = [
  {
    id: 1,
    label: 'Level one 1',
    children: [
      {
        id: 3,
        label: 'Level two 2-1',
        children: [
          {
            id: 4,
            label: 'Level three 3-1-1',
          },
          {
            id: 5,
            label: 'Level three 3-1-2',
            disabled: true,
          },
        ],
      },
      {
        id: 2,
        label: 'Level two 2-2',
        disabled: true,
        children: [
          {
            id: 6,
            label: 'Level three 3-2-1',
          },
          {
            id: 7,
            label: 'Level three 3-2-2',
            disabled: true,
          },
        ],
      },
    ],
  },
]

function flattenTree(data) {
  const result = []

  function flatten(node) {
    result.push(node)
    if (node?.children?.length) {
      node.children.forEach(flatten)
    }
  }

  data.forEach(flatten)
  return result
}

const checkedAll = (bool) => {
  console.log(bool, 'checkedAll')
  // 设置勾选节点  平铺的数据
  treeRef.value.setCheckedNodes(bool ? flattenTree(data).filter(item =>  !item.disabled ): [])
}
</script>

### What is Expected?

能够正常的点击

### What is actually happening?

需要点击两次复选框

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #15375

[playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiA8ZWwtY2hlY2tib3hcbiAgICAgICBAY2hhbmdlPVwiY2hlY2tlZEFsbFwiXG4gICAgICAgIHYtbW9kZWw9XCJjaGVja2VkVmFsXCJcbiAgICAgICAgPuWFqOmAiVxuICAgICAgPC9lbC1jaGVja2JveD5cbiAgPGVsLXRyZWVcbiAgICByZWY9XCJ0cmVlUmVmXCJcbiAgICA6ZGF0YT1cImRhdGFcIlxuICAgIHNob3ctY2hlY2tib3hcbiAgICBkZWZhdWx0LWV4cGFuZC1hbGxcbiAgICBub2RlLWtleT1cImlkXCJcbiAgICBoaWdobGlnaHQtY3VycmVudFxuICAgIDpwcm9wcz1cImRlZmF1bHRQcm9wc1wiXG4gIC8+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB0cmVlUmVmID0gcmVmKClcbmNvbnN0IGNoZWNrZWRWYWwgPSByZWYoZmFsc2UpXG5jb25zdCBkZWZhdWx0UHJvcHMgPSB7XG4gIGNoaWxkcmVuOiAnY2hpbGRyZW4nLFxuICBsYWJlbDogJ2xhYmVsJyxcbn1cblxuY29uc3QgZGF0YSA9IFtcbiAge1xuICAgIGlkOiAxLFxuICAgIGxhYmVsOiAnTGV2ZWwgb25lIDEnLFxuICAgIGNoaWxkcmVuOiBbXG4gICAgICB7XG4gICAgICAgIGlkOiA0LFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAxLTEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGlkOiA5LFxuICAgICAgICAgICAgbGFiZWw6ICdMZXZlbCB0aHJlZSAxLTEtMScsXG4gICAgICAgICAgfSxcbiAgICAgICAgICB7XG4gICAgICAgICAgICBpZDogMTAsXG4gICAgICAgICAgICBsYWJlbDogJ0xldmVsIHRocmVlIDEtMS0yJyxcbiAgICAgICAgICB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICB7XG4gICAgaWQ6IDIsXG4gICAgbGFiZWw6ICdMZXZlbCBvbmUgMicsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgaWQ6IDUsXG4gICAgICAgIGRpc2FibGVkOiB0cnVlLFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAyLTEnLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgaWQ6IDYsXG4gICAgICAgIGxhYmVsOiAnTGV2ZWwgdHdvIDItMicsXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG4gIHtcbiAgICBpZDogMyxcbiAgICBsYWJlbDogJ0xldmVsIG9uZSAzJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBpZDogNyxcbiAgICAgICAgbGFiZWw6ICdMZXZlbCB0d28gMy0xJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIGlkOiA4LFxuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAzLTInLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuXG5mdW5jdGlvbiBmbGF0dGVuVHJlZShkYXRhKSB7XG4gIGNvbnN0IHJlc3VsdCA9IFtdXG5cbiAgZnVuY3Rpb24gZmxhdHRlbihub2RlKSB7XG4gICAgcmVzdWx0LnB1c2gobm9kZSlcbiAgICBpZiAobm9kZT8uY2hpbGRyZW4/Lmxlbmd0aCkge1xuICAgICAgbm9kZS5jaGlsZHJlbi5mb3JFYWNoKGZsYXR0ZW4pXG4gICAgfVxuICB9XG5cbiAgZGF0YS5mb3JFYWNoKGZsYXR0ZW4pXG4gIHJldHVybiByZXN1bHRcbn1cblxuXG4gIGNvbnN0IGNoZWNrZWRBbGwgPSAoYm9vbCkgPT4ge1xuICAgIGNvbnN0IGxpc3QgPSBmbGF0dGVuVHJlZShkYXRhKS5maWx0ZXIoXG4gICAgICAgIChpdGVtKSA9PiAhaXRlbS5kaXNhYmxlZFxuICAgICAgKVxuICAgIC8vIOiuvue9ruWLvumAieiKgueCuSAg5bmz6ZO655qE5pWw5o2uXG4gICAgaWYgKGJvb2wpIHtcbiAgICAgIC8qICA7bGlzdC5mb3JFYWNoKChpdGVtKSA9PiB7XG4gICAgICAgIHRyZWVSZWYudmFsdWUuc2V0Q2hlY2tlZChpdGVtLCB0cnVlKVxuICAgICAgfSkqL1xuICAgICAgdHJlZVJlZi52YWx1ZS5zZXRDaGVja2VkTm9kZXMobGlzdClcbiAgICB9IGVsc2Uge1xuICAgICAgICB0cmVlUmVmLnZhbHVlLnNldENoZWNrZWROb2RlcyhbXSlcbiAgICB9XG4gIH1cblxuPC9zY3JpcHQ+XG4iLCJzcmMvUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTE1Mzc3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsInNyYy9lbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTE1Mzc3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ1xuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICB9KVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTE1Mzc3LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

### CODE REVIEW

**Review by ryuhangyeong — APPROVED** [review]

**Comment by ryuhangyeong** [comment]
@kooriookami 

Can you take a look at this?

File: packages/components/tree/src/model/tree-store.ts [hunk 1]
```diff
@@ -289,10 +289,23 @@ export default class TreeStore {
     leafOnly = false,
     checkedKeys: { [key: string]: boolean }
   ): void {
-    const allNodes = this._getAllNodes().sort((a, b) => b.level - a.level)
+    const allNodes = this._getAllNodes().sort((a, b) => {
+      if (a.level > b.level) {
+        return 1
+      }
+      return -1
+    })
```
Comment (kooriookami) [inline packages/components/tree/src/model/tree-store.ts:297]: Any different?

Comment (btea) [reply packages/components/tree/src/model/tree-store.ts:297]: The ones with level 1 before modification are ranked last, and those with level 1 after modification are ranked first.

Comment (kooriookami) [reply packages/components/tree/src/model/tree-store.ts:297]: const allNodes = this._getAllNodes().sort((a, b) => a.level - b.level);
What about this?

Comment (btea) [reply packages/components/tree/src/model/tree-store.ts:297]: Looks good, cleaner this way.

**Review by kooriookami — APPROVED** [review]

**Comment by ryuhangyeong** [comment]
@kooriookami 

Could you please check your email?

**Comment by kooriookami** [comment]
@ryuhangyeong Sure, I received your email, but I didn't receive your wechat verify.

**Comment by ryuhangyeong** [comment]
@kooriookami 

sorry. I just saw this answer. Is this possible now? If it is not possible within 5 minutes

I'll ask again tomorrow afternoon.

**Comment by kooriookami** [comment]
> @kooriookami
> 
> ~sorry. I just saw this answer. Is this possible now? If it is not possible within 5 minutes~
> 
> ~I'll ask again tomorrow afternoon.~
> 
> I'll ask you again tomorrow! 🙇

Korean time is 1 hour faster than China, you can tell me later what time the appointment is and if it's Korean time.

**Pull request merged**